File: medplum-java/medplum-java-model/src/main/java/com/fhir/ValueSet_Filter.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableValueSet_Filter.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ValueSet_Filter {
  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<String> value();

  Optional<Valueset_filterOp> op();

  Optional<code> property();

  static ImmutableValueSet_Filter.Builder builder() {
    return ImmutableValueSet_Filter.builder();
  }
}

File: medplum-typescript/src/model/fhir/ValueSet_Filter.ts
import { code } from "./code";
import { Extension } from "./Extension";
import { Valueset_filterOp } from "./Valueset_filterOp";

export interface ValueSet_Filter {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    property: code | null | undefined;
    op: Valueset_filterOp | null | undefined;
    value: string | null | undefined;
}
